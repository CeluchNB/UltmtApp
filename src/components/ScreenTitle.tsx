import * as React from 'react'
import { StyleSheet, Text, useColorScheme } from 'react-native'
import { darkColors, lightColors } from '../theme/colors'
import { size, weight } from '../theme/fonts'

const ScreenTitle: React.FC<{ title: string }> = ({ title }) => {
    const style = StyleSheet.create({
        title: {
            fontSize: size.fontLarge,
            color:
                useColorScheme() === 'dark'
                    ? darkColors.textPrimary
                    : lightColors.textPrimary,
            fontWeight: weight.bold,
            alignSelf: 'center',
            marginTop: 50,
        },
    })

    return <Text style={style.title}>{title}</Text>
}

export default ScreenTitle
