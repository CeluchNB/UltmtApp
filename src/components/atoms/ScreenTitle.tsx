import * as React from 'react'
import {
    StyleProp,
    StyleSheet,
    Text,
    TextStyle,
    useColorScheme,
} from 'react-native'
import { darkColors, lightColors } from '../../theme/colors'
import { size, weight } from '../../theme/fonts'

const ScreenTitle: React.FC<{
    title: string
    style?: StyleProp<TextStyle>
}> = ({ title, style }) => {
    const selfStyle = StyleSheet.create({
        title: {
            fontSize: size.fontLarge,
            color:
                useColorScheme() === 'dark'
                    ? darkColors.textPrimary
                    : lightColors.textPrimary,
            fontWeight: weight.bold,
        },
    })

    return <Text style={[selfStyle.title, style]}>{title}</Text>
}

export default ScreenTitle
