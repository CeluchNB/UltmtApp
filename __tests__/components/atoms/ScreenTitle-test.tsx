/**
 * @format
 */

import React from 'react'
import ReactNative from 'react-native'
import ScreenTitle from '../../../src/components/atoms/ScreenTitle'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

it('matches snapshot', () => {
    const tree = renderer.create(<ScreenTitle title="Login" />).toJSON()
    expect(tree).toMatchSnapshot()
})

it('renders correctly in dark mode', () => {
    const spy = jest
        .spyOn(ReactNative, 'useColorScheme')
        .mockImplementationOnce(() => 'dark')
    renderer.create(<ScreenTitle title="Login" />)
    spy.mockRestore()
})
