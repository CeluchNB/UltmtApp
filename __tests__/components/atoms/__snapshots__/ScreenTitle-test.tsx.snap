// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#3183ff",
        "fontSize": 82.1917808219178,
        "fontWeight": "bold",
      },
      undefined,
    ]
  }
>
  Login
</Text>
`;
